compiler/incremental-compilation-impl/src/org/jetbrains/kotlin/incremental/classpathDiff/ClasspathChangesComputer.kt

```diff
@@ -78,11 +78,31 @@ object ClasspathChangesComputer {
         }
 
         return metrics.measure(BuildTime.COMPUTE_IMPACTED_SET) {
-            // classChanges may contain added symbols (they can also impact recompilation -- see examples in JavaClassChangesComputer), so
-            // we need to consider all classes on both the current and previous classpath. It's okay for the combined list to contain
-            // overlapping ClassIds.
-            val allClasses = (currentClassSnapshots.asSequence() + changedPreviousClasses.asSequence()).asIterable()
-            computeImpactedSet(classChanges, allClasses)
+            // Find ProgramSymbols that are impacted by the changed ProgramSymbols.
```

Comment:
nit: Some the comments below seems like a duplication of the KDoc on the methods and duplication of the code that already have descriptive names.

Reply:
Done - I've revised the entire comment block.